application/models/team.py

```diff
@@ -0,0 +1,78 @@
+"""" Access to the Team model """
+import uuid
+from flask import current_app
+from py2neo import Node
+from application.errors import NotFound
+
+
+class Team:
+    """" Program representation of Team nodes in Neo4J """
+
+    @classmethod
+    def all(cls):
+        results = current_app.graph.run("MATCH (x:Team) RETURN x")
+        return list(map(
+            lambda result: Team(result['x']['name'], id_=result['x']['id']),
+            results
+        ))
+
+    @classmethod
+    def find_by(cls, force=False, id_='', name=''):
+        team_node = cls._find_team_node(id_=id_, name=name)
+        if team_node:
+            return Team(team_node['name'], id_=team_node['id'])
+        return cls._not_found(force)
+
+    @classmethod
+    def exists(cls, id_='', name=''):
+        return bool(cls._find_team_node(id_=id_, name=name))
+
+    @classmethod
+    def _not_found(cls, force=False):
+        if force:
+            raise NotFound()
+        else:
+            return None
+
+    @classmethod
+    def _find_team_node(cls, id_='', name=''):
+        query = f"MATCH (x:Team) WHERE x.name = '{name}' OR x.id = '{id_}' RETURN x"
+        return current_app.graph.run(query).evaluate()
+
```

Comment:
If you don't need the cls argument for helper methods it's better
to use `@staticmethod` where you don't pass `cls` as first parameter.

Reply:
If the method is static you can call it from an instance instead of on a class level. Which in this case wouldn't make any sense. I like containing the scope of this function this way.

Reply:
Conceptually a static method is a method that knows nothing about the class it could just as well exist without it's calls. You could even write it outside of the class scope. But this method doesn't make any sense at all without the team class which is why I think class method is more fitting personally.